packages/entity-list/src/main.js

```diff
@@ -105,12 +106,14 @@ class EntityListApp extends React.Component {
     this.app = initApp(props.id, props, events)
   }
 
-  componentWillReceiveProps = nextProps => {
-    const changedProps = _pickBy(nextProps, (value, key) => !_isEqual(value, this.props[key]))
-
-    getDispatchActions(changedProps).forEach(action => {
-      this.app.store.dispatch(action)
-    })
+  static getDerivedStateFromProps = props => {
+    const changedProps = _pickBy(props, (value, key) => !_isEqual(value, props[key]))
+    if (!_isEmpty(changedProps)) {
+      getDispatchActions(changedProps).forEach(action => {
+        this.app.store.dispatch(action)
+      })
+    }
+    return null
```

Comment:
this method actually never returns a result other than `null`. by intention?

I'm not familiar with using `getDerivedStateFromProps` but, as I understand it, this is not the use case this lifecycle method is meant for.

documentation (https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops):
"If you need to perform a side effect (for example, data fetching or an animation) in response to a change in props, use componentDidUpdate lifecycle instead."

Why did you choose `getDerivedStateFromProps` over `componentDidUpdate` in this case?

Reply:
I chose it because it gets called before render like `componentWillReceiveProps` to keep the timing of updates as close to the original method as possible. 
Return of at least some state change (in this case `null`) is required.